atlas-fastq-provider-functions.sh

```diff
@@ -1030,6 +1161,37 @@ convert_ena_fastq_to_ssh_path(){
     fi
 }
 
+convert_ena_fastq_to_fire_path(){
+    local fastq=$1
+    local status=${2:-'public'}
+    local library=${3:-''}
+    local returnCode=0
+
+    local fastq=$(basename $fastq)
+    if [ "$status" == 'private' ]; then
+        if [ "$library" == '' ]; then
+            echo "ERROR: For private FASTQ files, the library cannot be inferred from the file name and must be specified" 1>&2
+            returnCode=1
+        fi 
+    else
+        library=$(get_library_id_from_uri $fastq)
+        returnCode=$?
+    fi
+
+    if [ $returnCode -ne 0 ]; then
+        return $returnCode
+    else
+        local libDir=
+        if [ "$status" == 'private' ]; then
+            libDir=$(dirname $(get_library_path $library $ENA_PRIVATE_S3_ROOT_PATH 'short'))
```

Comment:
```suggestion
            libDir=$(dirname "$(get_library_path "$library" "$ENA_PRIVATE_S3_ROOT_PATH" 'short')")
```
double-quote the result to handle cases where the command output contains spaces or special characters.? Please check if that would work

Reply:
Addressed in 6ab945781b14d7ef3f15de066c8a36dcf695d9e0, also added this change in other similar commands throughout the script. Thanks!